toplevel/role/role.go

```diff
@@ -129,6 +139,9 @@ func (c config) Apply(entriesBytes []byte, dryRun bool, threadPoolSize int) {
 		}
 	}
 
+	addOptionalOidcDefaults(entries)
+	pruneUnsupported(entries)
```

Comment:
I'm not a huge fan of us having to do this, but this is may very well be the least bad way of doing things.

I'd like to think that either the vault server API, or the golang vault client library, are capable of filling in these defaults itself for us. But maybe that isn't a thing.

Reply:
Something to look into for the future perhaps.